src/main/webapp/app/course/competencies/components/course-competency-relation-form/course-competency-relation-form.component.ts

```
@@ -0,0 +1,289 @@
+            return this.getSelectableTailCompetencies(this.headCompetencyId()!, this.relationType()!);
+        }
+        return this.courseCompetencies();
+    });
+
+    private readonly currentAdjacencyMap = computed(() => this.getAdjacencyMap(this.relations()));
+
+    constructor() {
+        effect(
+            () => {
+                if (this.selectedRelationId()) {
+                    this.selectRelation(this.selectedRelationId()!);
```

Biome

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)